General in-place binary op support in dynamo

jbschlosser opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #94224
* __->__ #94203
* #94202

Continues the approach taken in #93271, expanding support to in-place binary ops (e.g. `__iadd__`).

cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire

jbschlosser committed:
General in-place binary op support in dynamo

[ghstack-poisoned]

ezyang commented on torch/_dynamo/variables/builtin.py:
```diff
@@ -257,6 +293,39 @@ def dynamic_handler(tx, a, b, options, fn=op):
+                ),
+            )
+
+        def tuple_iadd_handler(tx, a, b, options):
```
It's weird that tuple iadd needs to be handled explicitly, because this should just go to tuple add

jbschlosser replied:
You're right, lemme simplify

jbschlosser committed:
Update on "General in-place binary op support in dynamo"


Continues the approach taken in #93271, expanding support to in-place binary ops (e.g. `__iadd__`).

cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx desertfire

[ghstack-poisoned]
